htdocs/projet/document.php

```diff
@@ -122,7 +122,7 @@
 	print dol_get_fiche_head($head, 'document', $langs->trans("Project"), -1, ($object->public ? 'projectpub' : 'project'));
 
 	// Files list constructor
-	$filearray = dol_dir_list($upload_dir, "files", 0, '', '(\.meta|_preview.*\.png)$', $sortfield, (strtolower($sortorder) == 'desc' ?SORT_DESC:SORT_ASC), 1);
+	$filearray = dol_dir_list($upload_dir, "files",  !empty($conf->global->PROJECT_LIST_RECURSIVE_FILES), '', '(\.meta|_preview.*\.png)$', $sortfield, (strtolower($sortorder) == 'desc' ?SORT_DESC:SORT_ASC), 1);
```

Comment:
Adding option PROJECT_LIST_RECURSIVE_FILES bring a new independent feature. Must be pushed into an independent PR.